src/main/java/ch/uzh/ifi/hase/soprafs21/service/SchedulingService.java

```diff
@@ -0,0 +1,182 @@
+package ch.uzh.ifi.hase.soprafs21.service;
+
+import ch.uzh.ifi.hase.soprafs21.constant.SwipeStatus;
+import ch.uzh.ifi.hase.soprafs21.entities.*;
+import ch.uzh.ifi.hase.soprafs21.repository.ActivityRepository;
+import ch.uzh.ifi.hase.soprafs21.repository.ScheduledActivityRepository;
+import ch.uzh.ifi.hase.soprafs21.repository.SchedulingSessionRepository;
+import ch.uzh.ifi.hase.soprafs21.repository.UserRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.validation.ConstraintViolationException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Scheduling Service.
+ */
+@Service
+@Transactional
+public class SchedulingService {
+
+    private final Logger log = LoggerFactory.getLogger(SchedulingService.class);
+
+    private final SchedulingSessionRepository schedulingSessionRepository;
+
+    private final ScheduledActivityRepository scheduledActivityRepository;
+
+    private final ActivityRepository activityRepository;
+
+    private final UserRepository userRepository;
+
+    @Autowired
+    public SchedulingService(@Qualifier("schedulingSessionRepository") SchedulingSessionRepository schedulingSessionRepository,
+                             @Qualifier("scheduledActivityRepository") ScheduledActivityRepository scheduledActivityRepository,
+                             @Qualifier("activityRepository") ActivityRepository activityRepository,
+                             @Qualifier("userRepository") UserRepository userRepository) {
+        this.schedulingSessionRepository = schedulingSessionRepository;
+        this.scheduledActivityRepository = scheduledActivityRepository;
+        this.activityRepository = activityRepository;
+        this.userRepository = userRepository;
+    }
+
+    public SchedulingSession createSchedulingSession(Long userId1, Long userId2, String token) {
+        SchedulingSession schedulingSession = new SchedulingSession();
+        User user = userRepository.findByToken(token);
+        if (!user.getId().equals(userId1) && !user.getId().equals(userId2)) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "User can not start scheduling Session of other users.");
+        }
+
+        List<Activity> activityList = activityRepository.findAll();
+        List<Activity> sessionActivityList = new ArrayList<Activity>();
+        for (Activity activity : activityList) {
```

Comment:
Maybe creating a seperate method that handles this check (that checks if both users have swiped the given activities TRUE) would make the createSchedulingSession to be more compact, otherwise createSchedulingSession method seems to implement to much within itself?

Reply:
This check gets unnecessary after using your function getAllActivitiesWithMatchedUsers ;)